src/fides/api/ops/service/privacy_request/email_batch_service.py

```diff
@@ -160,62 +162,80 @@ def send_prepared_emails(
 
 
 @celery_app.task(base=DatabaseTask, bind=True)
-def send_consent_email_batch(self: DatabaseTask) -> ConsentEmailExitState:
-    """Sends consent emails for each relevant connector with
-    applicable user details batched together."""
+def send_email_batch(self: DatabaseTask) -> ConsentEmailExitState:
+    """Sends emails for each relevant connector with applicable user details batched together."""
 
-    logger.info("Starting batched consent email send...")
+    logger.info("Starting batched email send...")
     with self.get_new_session() as session:
 
         privacy_requests: Query = session.query(PrivacyRequest).filter(
-            PrivacyRequest.status == PrivacyRequestStatus.awaiting_consent_email_send
+            PrivacyRequest.status == PrivacyRequestStatus.awaiting_email_send
         )
         if not privacy_requests.first():
             logger.info(
-                "Skipping batch consent email send with status: {}",
+                "Skipping batch email send with status: {}",
                 ConsentEmailExitState.no_applicable_privacy_requests.value,
             )
             return ConsentEmailExitState.no_applicable_privacy_requests
 
-        conn_configs: Query = get_consent_email_connection_configs(session)
-        if not conn_configs.first():
-            requeue_privacy_requests_after_consent_email_send(privacy_requests, session)
+        consent_configs: Query = get_consent_email_connection_configs(session)
+        erasure_configs: Query = get_erasure_email_connection_configs(session)
+        combined_configs = consent_configs.union_all(erasure_configs)
+        if not combined_configs.first():
+            requeue_privacy_requests_after_email_send(privacy_requests, session)
             logger.info(
-                "Skipping batch consent email send with status: {}",
+                "Skipping batch email send with status: {}",
                 ConsentEmailExitState.no_applicable_connectors.value,
             )
             return ConsentEmailExitState.no_applicable_connectors
 
         batched_user_data: List[BatchedUserConsentData] = stage_resource_per_connector(
-            conn_configs
+            consent_configs
         )
         add_batched_user_preferences_to_emails(privacy_requests, batched_user_data)
```

Comment:
Can you add a code comment that these two items are consent-specific?   Or alternatively, at this point it seems like some of this consent-specific logic could be separated out into the file on the consent connector to better organize between what is consent-specific and what is generic. (similar to the ErasureConnector. send_erasure_email)

Reply:
I pulled the consent-specific logic into the `GenericConsentEmailConnector`